src/components/LayerSwitcher/AddCustomLayer.tsx

```diff
@@ -0,0 +1,283 @@
+      <span
+        style={{
+          display: 'flex',
+          gap: '1rem',
+          alignItems: 'center',
+        }}
+      >
+        {layer.icon && (
+          <img alt={`${layer.name}s logo`} src={layer.icon} height={20} />
+        )}
+        <h3>{layer.name}</h3>
+      </span>
+      {layer.description && <p>{layer.description}</p>}
+      {layer.category && <p>{layer.category}</p>}
+      <span
+        style={{
+          display: 'flex',
+          gap: '1rem',
+          alignItems: 'center',
+        }}
+      >
+        {layer.license_url && <a href={layer.license_url}>License</a>}
+        {layer.privacy_policy_url && (
+          <a href={layer.privacy_policy_url}>Privacy policy</a>
+        )}
+      </span>
+
+      <hr
+        style={{
+          width: '100%',
+        }}
+      />
+
+      {dynamicParts.map((part) => {
+        const hasOptions = !!part.options;
+
+        const prettyTitle =
+          {
+            switch: 'Tile server',
+            apikey: 'API-Key',
+          }[part.title] || part.title;
+
+        if (hasOptions)
+          return (
+            <Autocomplete
+              options={part.options}
+              renderInput={(params) => (
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                <TextField {...params} label={prettyTitle} />
+              )}
+              onChange={(_, val) => {
+                setValues((prev) => ({
+                  ...prev,
+                  [part.title]: val,
+                }));
+              }}
+              key={part.title}
+            />
+          );
+
+        return (
+          <TextField
+            label={prettyTitle}
+            key={part.title}
+            onChange={(val) => {
+              setValues((prev) => ({
+                ...prev,
+                [part.title]: val.target.value,
+              }));
+            }}
+          />
+        );
+      })}
+    </div>
+  );
+};
+
+interface AddDialogProps {
+  save: (layer: any) => void;
+}
+
+export const AddCustomDialog: React.FC<AddDialogProps> = ({ save }) => {
+  const { opened, close } = useAddLayerContext();
+  const [isSaveDisabled, setDisableSave] = React.useState(true);
+  const [page, setPage] = React.useState(0);
+
+  const [layer, setLayer] = React.useState<any | null>(null);
+  const [layerUrl, setLayerUrl] = React.useState<string | null>(null);
+
+  const onReset = () => {
+    close();
+    setPage(0);
+    setLayer(null);
+  };
+
+  const onSave = () => {
+    save({ ...layer, url: layerUrl });
+    onReset();
+  };
+
+  return (
+    <Dialog open={opened} maxWidth="sm" fullWidth>
+      <DialogTitle>
+        <p>Add a layer</p>
+      </DialogTitle>
+
+      <DialogContent>
+        {page === 0 && (
+          <LayerDataInput
+            onSelect={(newLayer) => {
+              setLayer(newLayer);
+              setDisableSave(!newLayer);
+            }}
+          />
+        )}
+        {page === 1 && (
+          <Details
```

Comment:
Can we perhaps move this second page just below the `LayerDataInput` autocomplete component? I think it would be easier for the user to explore, if they just select a imagery and imediatelly see the options without clicking on next.